### ISSUE TITLE
Per-site pass through of keybindings

### ISSUE THREAD

**Issue by ipod825** [Jelmerro/Vieb#57]
Many of google's page has their own mappings. For e.g. in gmail, you can configure it to press `j`, `k` to check next and previous email. In calendar, you can press `d`,`w`,`m`,`a` to switch between day/week/month/agenda mode. As vieb grabs all the keybindings, these site-dependent keybindings no longer works.

**Comment by Jelmerro** [comment]
One of the (many) core reasons for developing Vieb, for me, was the lack of separation between Vim-like keybindings and the website shortcuts. I have never regarded this merging of shortcuts as a positive, and therefor I haven't considered the possibility of adding this to Vieb.

Having said that, I'm not opposed to adding per-site pass-through of specific shortcuts. One of the things that isn't clear to me is how this would be configured in way that is in line with Vieb's current setting/mapping structure.

**Comment by ipod825** [comment]
Perhaps you can add a function called `passthroughkey`
```vim
nmap j call passthroughkey("j", "https://site-that-need-to-pass-through")
```

Implementation of `passthroughkey` would then be something like:
```vim
function! passthroughkey(key, site)
if match(current_url, site)
 " pass through a:key
else
  " handle key in vieb
endif
```

**Comment by Jelmerro** [comment]
I agree that this is a solution that would fit a Vim-like browser, but custom functions are not supported by Vieb as of today. Therefor we might want to consider a different implementation for now or we will need to wait for functions to be supported by Vieb before implementing this pass through feature.

**Comment by ipod825** [comment]
I think in addition to per-site pass-through. There should also be a global pass-through.
Regarding implementation complexity, I think these could be set using list, which is easier to parse.

```vim
let global_passthrough_keys = ['<Left>', '<Right>', '<Space>']
let per_site_passthrough_keys = [['j', _'*mail.google.com/*']]_
```

**Comment by Jelmerro** [comment]
I would like the set command to remain true to Vim, but we could introduce new commands to manage the passthough dictionary. I would probably stick to one dictionary and allow users to make global passthroughs by supplying an all-matching pattern as the domain.

**Comment by ipod825** [comment]
> I would like the set command to remain true to Vim

What does this mean? I thought mean also have options taking list/dict as argument.

> I would probably stick to one dictionary and allow users to make global passthroughs by supplying an all-matching pattern as the domain.

Better!

**Comment by Jelmerro** [comment]
I mean that this passthrough is more complex in configuration than any of the set command options. I would like the set command to remain as close to Vim as possible, and the set command is basically only for boolean toggles, numbers and strings (including comma-separated strings, which are used as lists). There aren't any options in Vim that behave like a dictionary, so I don't see why Vieb should have that (at least not as part of the set command). For this reason I prefer to use separate commands for this passthrough feature.

**Comment by ipod825** [comment]
I see. I recalled it wrong.

**Comment by Jelmerro** [comment]
Implemented in 11.0.0 :tada:
